roles/core/common/tasks/check.yml

```diff
@@ -87,13 +87,15 @@
     scale_daemon_nodename: "{{ scale_daemon_nodename }}"
   when: hostvars[inventory_hostname].scale_daemon_nodename is undefined
 
-- block:
-    - name: check | Get architecture
-      command: dpkg --print-architecture
-      register: ubuntu_arch
-      changed_when: false
-
-    - name: check | Set scale_architecture
-      set_fact:
-        scale_architecture: "{{ ubuntu_arch.stdout }}"
+# set dynamic variable based on supported OS
```

Comment:
General comment, have we looked into how these variables could be re-used depending on the calling roles.  If a role is called independently and not using the playbook.yml defined order...  do we have this common role as a dependency so that it gets executed and the variables that are required for other roles in the playbooks are set?

Reply:
Yes , I had added core/common role as dependency role in the meta for every supported component role , this is there from starting of the support . 

```
dependencies:
  - core/common
```

Reply:
That's the reason , i am setting global facts inside common role.  i have tested with all combination.